### PR TITLE
added TLS 1.2 tests for gateway sdk

### PR DESCRIPTION
**Pull request by rjaus**
Added a method & some tests for testing the SDK for TLS 1.2 usage.

See developer centre notice here: https://developer.xero.com/tls1-deprecation

These changes test against the Xero endpoint + an additional external API (https://www.howsmyssl.com/s/api.html) to determine that if TLS 1.2 is in use.

Open to comments / modifications / feedback

### CODE REVIEW

**Comment by rjaus** [comment]
Hey @nikz @armstrjare just an FYI, there will be a blog post going live today or tomorrow regarding TLS 1.2 (deprecating 1.0, encouraging 1.2).  Realise it's late notice, but it'd be great to have this (or an alternative solution) merged before then.

**Comment by nikz** [comment]
Heya @rjaus, thanks for the PR :)

I think we'd prefer not to commit tests that hit the internet, as that's generally considered bad practice.

Is the intent to give gem users a quick/easy way to test their networking stack and ensure it supports TLS 1.2? It looks like this support was [added to Ruby a while back](https://github.com/ruby/ruby/commit/060184c347822b11dff3db6bef915c04a564c4e4). Most people should be on 2.0 or later (as 1.9.x is officially unsupported).

tl;dr If you're on Ruby 2.0 or later you should be golden with TLS 1.2

**Comment by rjaus** [comment]
Hey @nikz, yes this was just an easy catch all to ensure everyone had upgraded.

Feel free to leave it out.

**Comment by nikz** [comment]
@rjaus sweet! Thanks for checking up - I'll ensure the min version for the gem is 2.0
